test/utils/post_utils.test.js

```diff
@@ -1079,5 +1085,33 @@ describe('PostUtils', () => {
             assert.equal(out.nextPosts[combinedPost.id].system_post_ids[3], '9');
             assert.deepEqual(out.nextPosts[combinedPost.id].props.user_activity, expectedUserActivityPosts);
         });
+
+        it('should combine consecutive combined and user activity posts between regular and custom posts', () => {
+            const out = combineSystemPosts(
+                [postId22, postIdUA12, postIdUA11, postCustomId2, postId1],
+                {[postIdUA11]: postUA11, [postIdUA12]: postUA12, [postId1]: post1, [postCustomId2]: postCustom2, [postId22]: post22},
+                'channel_id',
+            );
+            const expectedUserActivityPosts = {
+                allUserIds: ['user_id_2', 'added_user_id_1', 'user_id_1'],
+                messageData: [
+                    {postType: 'system_join_channel', userIds: ['user_id_2']},
+                    {postType: 'system_add_to_channel', userIds: ['added_user_id_1'], actorId: 'user_id_1'},
+                ],
+            };
+
+            const combinedPostId = out.postsForChannel[1];
+            console.log('out.nextPosts[combinedPostId]:', out.nextPosts[combinedPostId]);
+            console.log('out.postsForChannel:', out.postsForChannel);
```

Comment:
Are these logs needed or just for your debugging?

Reply:
Ooppss, now removed.